network/p2p/connManager.go

```diff
@@ -12,6 +12,16 @@ import (
 	"github.com/onflow/flow-go/module"
 )
 
+// TagLessConnManager is a companion interface to libp2p-core.connmgr.ConnManager which implements a (simplified) tagless variant of the Protect / Unprotect logic
+type TagLessConnManager interface {
+	connmgr.ConnManager
+	// ProtectPeer increments the stream setup count for the peer.ID
+	ProtectPeer(id peer.ID)
```

Comment:
The `ProtectPeer` and `UnprotectPeer` are only called when a one-to-one stream is established between two nodes (only caller us `libP2PNode.CreateStream -> tryCreateNewStream()` It is not called when a node connects to the other node as part of the PeerManager.
Hence, when testing nodes in a purely pub-sub connections, the `TagLessConnManager` observer will not be notified.

Libp2p nodes heartbeat every one second hence 2 seconds guarantee that peers have discovered each other and formed a mesh (their own tests do [something similar](https://github.com/libp2p/go-libp2p-pubsub/blob/master/gossipsub_test.go#L63))

If we want an event driven test and avoid these sleeps we could explore the libp2p `EventBus` - https://github.com/libp2p/go-libp2p-core/blob/master/event/network.go.

Reply:
Our tagless ProtectPeer and UnprotectPeer are indeed not called in pure pub-sub connections, but even if they were, they're not the calls that the TagWatchingConnManager intercepts and builds an Observable on.

What we are capturing here is the calls made by [libp2p's tagtracer in the pubsub package](https://github.com/libp2p/go-libp2p-pubsub/blob/master/tag_tracer.go). They call the tag-full Protect (and Unprotect), not the tag-less ones.

Reply:
ah ok ! got it..thanks